File: backend/src/main/java/com/zssystem/entity/Department.java
package com.zssystem.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("sys_department")
public class Department {
    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("parent_id")
    private Long parentId;

    @TableField("dept_name")
    private String deptName;

    @TableField("dept_code")
    private String deptCode;

    private String leader;

    private String phone;

    private String email;

    @TableField("sort_order")
    private Integer sortOrder;

    private Integer status; // 0-禁用，1-启用

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/department.ts
import request from '@/utils/request';

export interface DepartmentTreeVO {
  id: number;
  parentId: number;
  deptName: string;
  deptCode?: string;
  leader?: string;
  phone?: string;
  email?: string;
  sortOrder: number;
  status: number;
  createTime?: string;
  updateTime?: string;
  children?: DepartmentTreeVO[];
}

export interface DepartmentSaveParams {
  id?: number;
  parentId?: number;
  deptName: string;
  deptCode?: string;
  leader?: string;
  phone?: string;
  email?: string;
  sortOrder?: number;
  status?: number;
}

// 获取部门树
export const getDepartmentTree = () => {
  return request.get<DepartmentTreeVO[]>('/department/tree');
};

// 获取部门详情
export const getDepartmentById = (id: number) => {
  return request.get<DepartmentTreeVO>(`/department/${id}`);
};

// 创建部门
export const createDepartment = (data: DepartmentSaveParams) => {
  return request.post<void>('/department', data);
};

// 更新部门
export const updateDepartment = (id: number, data: DepartmentSaveParams) => {
  return request.put<void>(`/department/${id}`, data);
};

// 删除部门
export const deleteDepartment = (id: number) => {
  return request.delete<void>(`/department/${id}`);
};
